cmake/modules/DetectMachine.cmake

```diff
@@ -20,22 +20,40 @@ ENDIF()
 MESSAGE("PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
 SET(LMMS_HOST_X86 FALSE)
 SET(LMMS_HOST_X86_64 FALSE)
+SET(LMMS_HOST_ARM32 FALSE)
+SET(LMMS_HOST_ARM64 FALSE)
+SET(LMMS_HOST_RISCV32 FALSE)
+SET(LMMS_HOST_RISCV64 FALSE)
 
 IF(NOT DEFINED WIN64 AND CMAKE_SIZEOF_VOID_P EQUAL 8)
-    SET(WIN64 ON)
+	# TODO: This seems a bit presumptous
+	SET(WIN64 ON)
 ENDIF()
 
 IF(WIN32)
-	IF(WIN64)
-		SET(IS_X86_64 TRUE)
-		SET(LMMS_BUILD_WIN64 TRUE)
-	ELSE(WIN64)
-		SET(IS_X86 TRUE)
-	ENDIF(WIN64)
-
 	if(MSVC)
 		SET(MSVC_VER ${CMAKE_CXX_COMPILER_VERSION})
 
+		# Detect target architecture
+		IF(CMAKE_GENERATOR_PLATFORM)
+			STRING(TOLOWER "${CMAKE_GENERATOR_PLATFORM}" MSVC_TARGET_PLATFORM)
+		ELSE()
+			STRING(TOLOWER "${CMAKE_VS_PLATFORM_NAME_DEFAULT}" MSVC_TARGET_PLATFORM)
+		ENDIF()
+
+		IF(MSVC_TARGET_PLATFORM MATCHES "x64")
+			SET(IS_X86_64 TRUE)
+			SET(LMMS_BUILD_WIN64 TRUE)
```

Comment:
Is `LMMS_BUILD_WIN64` only for x86-64 builds, not for ARM64?

Reply:
Until the build system has been tested against `llvm-mingw` (and all mingw dependencies are cross-compiled, hosted at a PPA) or `vcpkg` is tested using Windows ARM64, this is up in the air.  Personally, I'd discourage obfuscating the two.  Historically `WIN64` and `LMMS_BUILD_WIN64 ` were synonymous with "x86_64".  Moving forward, this is up in the air for the following reasons:
* Windows ARM64 can execute x86_64 binaries
* Windows ARM64 can execute x86 binaries
* Windows ARM64 can execute ARM64 binaries

For these reasons, RemoteVstPlugin64, RemoveVstPlugin32 may need adjustments so that Windows 11 on ARM64 can run 32-bit and 64-bit Intel binaries.

Perhaps a better question is, "Will these changes break our existing MSVC builds?" and I think they will not, and they are preparing us in part for a day when we can add support for ARM64 on Windows as well.  The `MESSAGE()` alone helps considerably when debugging why a MSVC build isn't firing.

Reply:
Note, I can offer Windows 11 ARM64 hardware if someone is interested in improving the above.